user/tests/wiring/power_management/power.cpp

```diff
@@ -530,6 +594,20 @@ test(POWER_15_DisableFeatureFlag) {
         delay(5000);
         notifyAndReset();
     } else if (g_state == STATE11_MAGICK) {
+        auto cfg = System.getPowerConfiguration();
+        assertEqual(cfg.isFeatureSet(SystemPowerFeature::PMIC_DETECTION), false);
+#if HAL_PLATFORM_POWER_WORKAROUND_USB_HOST_VIN_SOURCE
+        assertEqual(cfg.isFeatureSet(SystemPowerFeature::USE_VIN_SETTINGS_WITH_USB_HOST), true);
+#else
+        assertEqual(cfg.isFeatureSet(SystemPowerFeature::USE_VIN_SETTINGS_WITH_USB_HOST), false);
+#endif // HAL_PLATFORM_POWER_WORKAROUND_USB_HOST_VIN_SOURCE
+        assertEqual(cfg.isFeatureSet(SystemPowerFeature::DISABLE), true);
+        assertEqual(cfg.isFeatureSet(SystemPowerFeature::DISABLE_CHARGING), false);
+        assertEqual(cfg.getPowerSourceMaxCurrent(), particle::power::DEFAULT_INPUT_CURRENT_LIMIT);
+        assertEqual(cfg.getPowerSourceMinVoltage(), particle::power::DEFAULT_INPUT_VOLTAGE_LIMIT);
+        assertEqual(cfg.getBatteryChargeCurrent(), particle::power::DEFAULT_CHARGE_CURRENT);
+        assertEqual(cfg.getBatteryChargeVoltage(), particle::power::DEFAULT_TERMINATION_VOLTAGE);
+
         assertEqual(System.batteryState(), (int)BATTERY_STATE_UNKNOWN);
         assertEqual(System.powerSource(), (int)POWER_SOURCE_UNKNOWN);
         // Restore default power management configuration
```

Comment:
TODO: (can be a separate PR) let's add three tests 1) disable/enable charging in runtime and verifying configuration and actual PMIC state 2) disable/enable charging, resetting and making sure that the necessary charging state is set on boot 3) disable/enable chargin, triggering PMIC watchdog and making sure that the necessary charging state is achieved after watchdog timer expires.

Reply:
Yeah, I'm still testing some of this code so I didn't add the tests here yet.